onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,102 @@
+"""
+Delete users management command.
+"""
+import sys
+from django.contrib.auth.models import User
+from onadata.apps.logger.models import XForm
+from onadata.apps.logger.models import Instance
+from onadata.apps.logger.models import Project
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import timezone
+
+
+def get_user_object_stats(username):  # pylint: disable=R0201
+    """
+    Get User information.
+    """
+    user = User.objects.get(username=username)
+    # Get the number of projects for this user
+    user_projects = Project.objects.filter(created_by=user).count()
+    # Get the number of forms
+    user_forms = XForm.objects.filter(user=user).count()
+    # Get the number of submissions
+    user_sumbissions = Instance.objects.filter(user=user).count()
+
+    result = {
+        'projects': user_projects,
+        'forms': user_forms,
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write(
+            sys.style.SUCCESS(
+                'User {} deleted successfully.'.format(username)))
+        # confirm too that no user exists with provided email
+        if len(User.objects.filter(email=user.email, is_active=True)) > 1:
+            inactivate_user(user.username)
```

Comment:
Is there a chance of the same email being present in another account ? _I don't think there is.... but if there is a possibility for that, there is a chance that we might delete and unrelated user_

Reply:
Once again not entirely sure if this is necessary ? Is there a chance of the actions made in LN44-48 failing ?

_I also believe that with this line of code added a read replica db setup for onadata may act weirdly here...._

Reply:
For instance when a user created a test account with `wkiragu+1@ona.io` does that appy as the same email as `wkiragu@ona.io`?

Reply:
Nope... That shouldn't be the same account

Reply:
I will leave that there as an extra check to confirm then that no other user account exists with the same email, maybe not inactivate the other accounts, I've changed this to alert the user running the command of this
I think it will be a good to have. thoughts?

Reply:
I think that's good 👍